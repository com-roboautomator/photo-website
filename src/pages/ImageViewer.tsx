import {createStyles, withStyles, WithStyles} from '@material-ui/styles'
import React from 'react'
import {data} from '../assets/data/TestData'

interface ImageViewerProps extends WithStyles<typeof styles> {
    source?: string
}

class ImageViewer extends React.Component<ImageViewerProps, any> {
    constructor(props: ImageViewerProps) {
        super(props)

        this.state = {
            properties: data,
        }
    }

    render() {
        const {properties} = this.state
        const property: string = properties.find(
            (x: {url: string}) =>
                x.url ===
                'https://i.picsum.photos/id/158/4836/3224.jpg?hmac=Gu_3j3HxZgR74iw1sV0wcwlnSZSeCi7zDWLcjblOp_c'
        ).url
        const classes = this.props.classes

        return (
            <div>
                <img src={property} alt="nata" className={classes.container} />
            </div>
        )
    }
}

const styles = () =>
    createStyles({
        container: {
            display: 'block',
            width: '50%',
            //height: '50%',
            aspectRatio: '1',
        },
    })

export default withStyles(styles)(ImageViewer)
