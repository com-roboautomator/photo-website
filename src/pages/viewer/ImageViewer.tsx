import {createStyles, withStyles, WithStyles} from '@material-ui/styles'
import React from 'react'
import {RouteComponentProps} from 'react-router-dom'
import {
    collection,
    image,
    placeholder,
} from '../../assets/data/ImageDataStructure'
import ImageViewerTitle from './ImageViewerTitle'
import ImageViewerViewport from './ImageViewerViewport'
import ImageViewerIndex from './ImageViewerIndex'
import ImageViewerDescription from './ImageViewerDescription'
import {settings} from '../../assets/settings.json'

interface ImageViewerProps
    extends WithStyles<typeof styles>,
        React.PropsWithChildren<RouteComponentProps<any, any, unknown>> {}

interface ImageViewerState {
    collection: collection
    index: number
    previousIndex: number
}

class ImageViewer extends React.Component<ImageViewerProps, ImageViewerState> {
    componentDidMount = () => {
        fetch(
            settings.domain +
                settings.port +
                '/collection/' +
                this.props.match.params.collectionId,
            {
                method: 'get',
                headers: {
                    'Content-Type': 'application/json',
                },
            }
        )
            .then((response) => response.json())
            .then((response) => {
                console.log(response)
                this.setState({
                    collection: response,
                })
            })
    }

    constructor(props: ImageViewerProps) {
        super(props)
        this.state = {
            index: 0,
            previousIndex: 0,
            collection: {
                id: 'null key',
                images: [],
                title: 'null title',
                index: 0,
            },
        }
    }

    next = () => {
        const currentIndex = this.state.index
        const updatedIndex = this.state.index + 1
        this.setState({
            previousIndex: currentIndex,
            index: updatedIndex,
        })
    }

    previous = () => {
        const currentIndex = this.state.index
        const updatedIndex = this.state.index - 1
        this.setState({
            previousIndex: currentIndex,
            index: updatedIndex,
        })
    }

    render() {
        if (
            this.state.collection === null ||
            this.state.collection.images.length === 0
        ) {
            console.log('returning')
            return <div />
        }

        for (let x = 0; x < this.state.collection.images.length; x++) {
            this.setState({index: x})
        }

        const collection = this.state.collection
        const images = collection?.images

        const state = this.state
        const classes = this.props.classes

        const collectionLength =
            collection?.images === undefined ? 0 : collection.images.length
        const targetImage: image =
            collection?.images === undefined
                ? placeholder
                : collection?.images[state.index]

        return (
            <div className={classes.container}>
                <ImageViewerTitle
                    collectionTitle={collection?.title.toUpperCase() ?? ''}
                    imageTitle={
                        images === undefined
                            ? ''
                            : images[this.state.index].title.toUpperCase() ?? ''
                    }
                />
                <ImageViewerViewport
                    index={this.state.index}
                    collection={this.state.collection}
                    next={this.next}
                    previous={this.previous}
                />
                <ImageViewerIndex
                    indexValue={this.state.index}
                    indexTotal={collectionLength}
                />
                <ImageViewerDescription
                    title={targetImage.title}
                    text={targetImage.description ?? ''}
                />
            </div>
        )
    }
}

const styles = () =>
    createStyles({
        container: {
            display: 'flex',
            flexDirection: 'column',

            justifyContent: 'center',
            alignContent: 'center',
        },
    })

export default withStyles(styles)(ImageViewer)
