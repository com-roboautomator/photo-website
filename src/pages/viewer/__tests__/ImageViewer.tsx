import '@testing-library/jest-dom/extend-expect'
import {render} from '@testing-library/react'
import renderer from 'react-test-renderer'
import {
    BrowserRouter,
    Route,
    Router,
    useHistory,
    useParams,
} from 'react-router-dom'
import ImageViewer from '../ImageViewer'

describe('Image Viewer', () => {
    it('renders correctly when there are no items', () => {
        const navigationBarRender = renderer
            .create(
                <BrowserRouter>
                    <Route
                        render={(props: any) => <ImageViewer {...props} />}
                    />
                </BrowserRouter>
            )
            .toJSON()

        expect(navigationBarRender).toMatchSnapshot()
    })

    it('should render correctly', () => {
        const {getByTestId} = render(
            <BrowserRouter>
                <Route render={(props) => <ImageViewer {...props} />} />
            </BrowserRouter>
        )

        expect(
            getByTestId('Image-Viewer-Title-Collection-Title')
        ).toBeInTheDocument()
        expect(
            getByTestId('Image-Viewer-Viewport-Previous-Image')
        ).toBeInTheDocument()
        expect(getByTestId('Image-Viewer-Index')).toBeInTheDocument()
        expect(
            getByTestId('Image-Viewer-Description-Title')
        ).toBeInTheDocument()
        expect(getByTestId('Image-Viewer-Description-Text')).toBeInTheDocument()
    })
})
