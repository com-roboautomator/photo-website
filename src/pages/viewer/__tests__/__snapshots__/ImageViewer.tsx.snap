// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Viewer renders correctly when there are no items 1`] = `
<div
  className="ImageViewer-container-1"
>
  <div
    className="ImageViewerTitle-container-2"
  >
    <div
      className="ImageViewerTitle-collection_title-3"
      data-testid="Image-Viewer-Title-Collection-Title"
    >
      
    </div>
    <div
      className="ImageViewerTitle-image_title-4"
      data-testid="Image-Viewer-Title-Image-Title"
    >
      
    </div>
  </div>
  <div
    className="ImageViewerViewport-container-5"
  >
    <img
      alt=""
      className="ImageViewerViewport-previous-8"
      data-testid="Image-Viewer-Viewport-Previous-Image"
      src=""
    />
    <img
      alt=""
      className="ImageViewerViewport-image-7"
      data-testid="Image-Viewer-Viewport-Target-Image"
      src=""
    />
    <div
      className="ImageViewerViewport-button_wrapper-9"
      id="Image-Viewer-Viewport-Button-Wrapper"
    >
      <div
        className="Arrow-container-10"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text Mui-disabled Mui-disabled"
          data-testid="Arrow-Button-Left"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <img
              alt=""
              data-testid="Arrow-Button-Image"
              src="arrow-disabled.png"
              style={
                Object {
                  "transform": "rotate(180deg)",
                }
              }
              width="40"
            />
          </span>
        </button>
      </div>
      <div
        className="Arrow-container-10"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text Mui-disabled Mui-disabled"
          data-testid="Arrow-Button-Right"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <img
              alt=""
              data-testid="Arrow-Button-Image"
              src="arrow-disabled.png"
              style={
                Object {
                  "transform": "rotate(0deg)",
                }
              }
              width="40"
            />
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="ImageViewerIndex-container-11"
    data-testid="Image-Viewer-Index"
  >
    1 / 0
  </div>
  <div
    className="ImageViewerDescription-container-12"
  >
    <div
      className="ImageViewerDescription-title-13"
      data-testid="Image-Viewer-Description-Title"
    >
      Placeholder
    </div>
    <div
      className="ImageViewerDescription-text-14"
      data-testid="Image-Viewer-Description-Text"
    >
      
    </div>
  </div>
</div>
`;
